clippy_lints/src/non_copy_const.rs

```diff
@@ -34,6 +34,14 @@ declare_clippy_lint! {
     /// `std::sync::ONCE_INIT` constant). In this case the use of `const` is legit,
     /// and this lint should be suppressed.
     ///
+    /// When an enum has variants with interior mutability, use of its non interior mutable
+    /// variants can generate false positives. See issue [#3962](https://github.com/rust-lang/rust-clippy/issues/3962)
+    ///
+    /// Types that have underlying or potential interior mutability trigger the lint, but are
+    /// usually not the type of errors that the lint is trying to catch. See issues
+    /// [#5050](https://github.com/rust-lang/rust-clippy/issues/5050),
+    /// [#5812](https://github.com/rust-lang/rust-clippy/issues/5812)
```

Comment:
Please also add these comments to the `borrow_...` lint

Additionally the issue #3825 is relevant here.